c_binding/lsm_local_disk.c

```diff
@@ -781,19 +782,20 @@ static int _sysfs_sas_addr_get(const char *blk_name, char *tp_sas_addr)
         goto out;
 
     sprintf(sysfs_sas_path, "/sys/block/%s/device/sas_address", blk_name);
+    if (! _file_exists(sysfs_sas_path))
+        goto out;
+
+    tmp_rc = _read_file(sysfs_sas_path, (uint8_t *) sysfs_sas_addr, &read_size,
+                        _SYSFS_SAS_ADDR_LEN);
+    /* As sysfs entry has trailing '\n', we should get EFBIG here */
```

Comment:
Any sysfs entry with contents has extra trailing '\n', but the way we call _read_file() does not want the trailing '\n'. Let me explained with example:

```
[fge@megadeth ~]$ cat /sys/block/sdr/device/sas_address 
0x500015535990a200
```
We have this SAS address sysfs entry with 16 hex digits and `0x` and trailing '\n' and '\0', so the total length is 20. Previously we call `_read_file()` to request only 19 bytes and replace the last '\n' with '\0', but with f786fa4, the `_read_file()` will fail in this usage with `EFBIG`, then similar usage all failed.

This also apply to `_read_file()` in `_ses_sg_paths_get()` which means no SES sg will be chose, hence all SES functions down.